### PR TITLE
Dni 7 k drivers linux5 10 port

### PR DESCRIPTION
**Pull request by eladnachman**
Pull Request for DNI Armada 7K drivers support for Linux 5.10 kernel

### CODE REVIEW

**Review by taraschornyiplv — APPROVED** [review]

File: packages/base/any/kernels/5.10-lts/patches/0024-i2c-Add-mv64xxx-support-for-AXI-bus-clock-handling.patch [hunk 1]
```diff
@@ -0,0 +1,109 @@
+From 5604c64109bcd315c5d2e5461b7304f361d93642 Mon Sep 17 00:00:00 2001
```
Comment (paulmenzel) [inline packages/base/any/kernels/5.10-lts/patches/0024-i2c-Add-mv64xxx-support-for-AXI-bus-clock-handling.patch:1]: Where is that commit upstream?

    $ git show 5604c64109bcd315c5d2e5461b7304f361d93642
    fatal: bad object 5604c64109bcd315c5d2e5461b7304f361d93642

File: packages/base/any/kernels/5.10-lts/patches/0024-i2c-Add-mv64xxx-support-for-AXI-bus-clock-handling.patch [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+ 	wait_queue_head_t	waitq;
+ 	spinlock_t		lock;
+ 	struct i2c_msg		*msg;
+@@ -908,6 +909,14 @@ mv64xxx_i2c_probe(struct platform_device *pd)
+ 	if (!IS_ERR(drv_data->reg_clk))
+ 		clk_prepare_enable(drv_data->reg_clk);
+ 
++	drv_data->axi_clk = devm_clk_get(&pd->dev, "axi");
++	if (IS_ERR(drv_data->axi_clk) &&
++	    PTR_ERR(drv_data->axi_clk) == -EPROBE_DEFER) {
```
Comment (paulmenzel) [inline packages/base/any/kernels/5.10-lts/patches/0024-i2c-Add-mv64xxx-support-for-AXI-bus-clock-handling.patch:82]: Please follow the Linux kernel coding style, and run it through `checkpatch.pl`.

Comment (eladnachman) [reply packages/base/any/kernels/5.10-lts/patches/0024-i2c-Add-mv64xxx-support-for-AXI-bus-clock-handling.patch:82]: ./scripts/checkpatch.pl /dentOS/packages/base/any/kernels/5.10-lts/patches/0024-i2c-Add-mv64xxx-support-for-AXI-bus-clock-handling.patch
total: 0 errors, 0 warnings, 72 lines checked
dentOS/packages/base/any/kernels/5.10-lts/patches/0024-i2c-Add-mv64xxx-support-for-AXI-bus-clock-handling.patch has no obvious style problems and is ready for submission.

Comment (paulmenzel) [reply packages/base/any/kernels/5.10-lts/patches/0024-i2c-Add-mv64xxx-support-for-AXI-bus-clock-handling.patch:82]: Ah, sorry, it’s the continuation from the if condition.

File: packages/base/any/kernels/5.10-lts/patches/0027-net-Add-Ethernet-2500Base-T-support-for-Marvell-10G.patch [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+--- a/drivers/net/phy/marvell10g.c
++++ b/drivers/net/phy/marvell10g.c
+@@ -263,10 +263,6 @@ static int mv3310_power_up(struct phy_device *phydev)
+ 	ret = phy_clear_bits_mmd(phydev, MDIO_MMD_VEND2, MV_V2_PORT_CTRL,
+ 				 MV_V2_PORT_CTRL_PWRDOWN);
+ 
+-	if (phydev->drv->phy_id != MARVELL_PHY_ID_88X3310 ||
+-	    priv->firmware_ver < 0x00030000)
+-		return ret;
+-
```
Comment (paulmenzel) [inline packages/base/any/kernels/5.10-lts/patches/0027-net-Add-Ethernet-2500Base-T-support-for-Marvell-10G.patch:26]: Why is this removed?

Comment (eladnachman) [reply packages/base/any/kernels/5.10-lts/patches/0027-net-Add-Ethernet-2500Base-T-support-for-Marvell-10G.patch:26]: This is an old limitation. Today there is a much greater variety of 10G copper PHY chips by Marvell today than just the 88X3310. Please see the updated list here: https://www.marvell.com/products/ethernet-phys.html .

Comment (paulmenzel) [reply packages/base/any/kernels/5.10-lts/patches/0027-net-Add-Ethernet-2500Base-T-support-for-Marvell-10G.patch:26]: Please make it a separate commit, and update the commit message. When are you going to send it upstream?

**Comment by paulmenzel** [comment]
The commit should be in the upstream Linux kernel, so they get properly reviewed. Have you done this already?

**Comment by sonoble** [comment]
Hi @Mickey201 is it possible to get these signed off? We have been merging improperly signed items for a while and I don't believe we should be continuing to do so.

**Comment by paulmenzel** [comment]
@sonoble, without upstreaming effort maintaining the Linux kernel is going to be harder. Please let’s not repeat the ONL mistakes from the past.

**Comment by jmpolom** [comment]
I'll second this sentiment. Without a full blown kernel team to maintain a kernel fork, this is a losing battle. Based on everything I've seen the DENT project does not have the resources to maintain an out of tree kernel fork.
